R/method-differentialAbundance.R

```diff
@@ -0,0 +1,170 @@
+#' Differential abundance
+#'
+#' This function returns the fold change and associated p value for a differential abundance analysis comparing samples in two groups.
+#' 
+#' @param data AbundanceData object
+#' @param comparisonVariable string identifying the metadata column to be used when grouping samples.
+#' @param groupA array of strings, each string matching a metadata value. Samples with this metadata value will be included in Group A
+#' @param groupB array of strings, each string matching a metadata value. Samples with this metadata value will be included in Group B. Must be distinct from groupA.
+#' @param method string defining the the differential abundance method. Accepted values are 'DESeq'
+#' @param verbose boolean indicating if timed logging is desired
+#' @return ComputeResult object
+#' @import veupathUtils
+#' @import data.table
+#' @useDynLib microbiomeComputations
+#' @export
+setGeneric("differentialAbundance",
+  function(data, comparisonVariable, groupA, groupB, method = c('DESeq'), verbose = c(TRUE, FALSE)) standardGeneric("differentialAbundance"),
+  signature = c("data")
+)
+
+#'@export
+setMethod("differentialAbundance", signature("AbundanceData"), function(data, comparisonVariable, groupA, groupB, method = c('DESeq'), verbose = c(TRUE, FALSE)) {
```

Comment:
worth considering a class for the three comparison related args?

Reply:
yes i like this a lot